Sources/SwiftInspectorVisitors/Tests/TypeDescriptionSpec.swift

```diff
@@ -684,6 +731,51 @@ final class TypeDescriptionSpec: QuickSpec {
           expect(visitor?.classRestrictionIdentifier?.asSource) == "AnyObject"
         }
       }
+
+      context("when called on a TypeSyntax node representing a FunctionTypeSyntax") {
+        final class FunctionTypeSyntaxVisitor: SyntaxVisitor {
+          var functionIdentifier: TypeDescription?
+          // Note: ideally we'd visit a node of type FunctionTypeSyntax
+          // but there's no way to get a TypeSyntax from an object of that type.
+          override func visit(_ node: TypeAnnotationSyntax) -> SyntaxVisitorContinueKind {
+            functionIdentifier = node.type.as(TypeSyntax.self)?.typeDescription
+            return .skipChildren
+          }
+        }
+
+        context("on a function that does not throw") {
+          var visitor: FunctionTypeSyntaxVisitor!
+          beforeEach {
+            let content = """
+                var test: (Int, Double) -> String
```

Comment:
Do we need to handle `@escaping`?

Reply:
Down the road we will also need to handle `async`.

Reply:
We will need to handle `async`, but `FunctionTypeSyntax` [doesn't actually have a way to access `@escaping`](https://github.com/apple/swift-syntax/blob/ad768ea60c913afc15bab05e3a3d2ef77a46beba/Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxTypeNodes.swift#L1442). I'm not sure what syntax type captures that information.